meta-client/src/test/java/org/apache/helix/metaclient/puppy/AbstractPuppy.java

```diff
@@ -0,0 +1,95 @@
+package org.apache.helix.metaclient.puppy;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.metaclient.api.MetaClientInterface;
+import java.util.HashMap;
+
+/**
+ * AbstractPuppy object contains interfaces to implement puppy and main logics to manage puppy life cycle
+ */
+public abstract class AbstractPuppy implements Runnable {
+
+  protected MetaClientInterface<String> _metaclient;
+  protected PuppySpec _puppySpec;
+  public HashMap<String, Integer> _eventChangeCounterMap;
+  protected int _unhandledErrorCounter;
+
+  public AbstractPuppy(MetaClientInterface<String> metaclient, PuppySpec puppySpec) {
+    this._metaclient = metaclient;
+    this._puppySpec = puppySpec;
```

Comment:
I guess no need for 'this'?

Reply:
My bad I missed it! Thanks for the catch